### PR TITLE
new prefix for images produced in testing: baseline_ or thisPR_

### PR DESCRIPTION
**Pull request by nastasha-w**
## PR Summary

I'm adding prefixes to the file names produced in `yt/yt/utilities/answer_testing/framework.py`, `dump_images` function. This should make it straightforward to see which image is the baseline for the comparison and which has been produced by the pull request branch in the `nose` testing output.

This addresses issue #4954.

### CODE REVIEW

**Review by neutrinoceros — CHANGES_REQUESTED** [review]
https://github.com/yt-project/yt/pull/4955#pullrequestreview-2197370110 👀

**Comment by neutrinoceros** [comment]
You can't just revert these commits by stacking more commits, these files need to be obliterated from the history.

**Comment by nastasha-w** [comment]
@neutrinoceros I think I've fixed it for this specific branch; please let me know if it's not ok.

**Review by neutrinoceros — APPROVED** [review]

**Comment by neutrinoceros** [comment]
Before we merge this, it'd be great to check if it actually works as we think it does. Can you alter some image answer tests so we get to see a diff report ? We'll remove the temporary commit afterwards

**Comment by nastasha-w** [comment]
I was hoping the tests on PR #4934 would show that, since those image tests are failing anyway. It's why I merged these changes into that branch.

**Comment by neutrinoceros** [comment]
I believe you mean #4939

**Comment by nastasha-w** [comment]
> I believe you mean #4939

Yup, thanks! I edited it.

**Comment by nastasha-w** [comment]
It seems to work! The file names in the latest tests for PR #4939 ([link](https://tests.yt-project.org/job/yt_py310_git/8364/#showFailuresLink)) are e.g.
'baseline_jkwqzihf.png'
'thisPR_4d5dlaqe.png'
'thisPR_4d5dlaqe-failed-diff.png'

**Pull request merged**
